src/components/ec/cuda/kernel/ec_cuda_executor.cu

```diff
@@ -253,6 +258,77 @@ __device__ void executor_copy_multi(ucc_eee_task_copy_multi_t *task)
     }
 }
 
+__global__ void
+executor_kernel_cooperative(volatile ucc_ec_cuda_executor_t *eee, int q_size)
```

Comment:
pls unify executor_kernel_cooperative and executor_kernel, they are almost identical